packages/good-design/src/apps/onramp/Onramper.tsx

```diff
@@ -0,0 +1,176 @@
+      </CentreBox>
+    </HStack>
+  );
+};
+
+const Stepper = ({ step = -1 }) => (
+  <VStack direction={"row"} mb={6} justifyContent="center" justifyItems="center" position="relative">
+    <StepsProgress step={step} />
+    <HStack alignItems="center" h="70px" justifyContent="space-between" w="300">
+      <Stack width={"1/3"} alignItems={"center"}>
+        <Circle size="40px" bgColor={step >= 0 ? "primary" : "goodGrey.300"}>
+          <Text color={step >= 0 ? "white" : "goodGrey.700"}> 1 </Text>
+        </Circle>
+        <Text color="goodGrey.700" fontFamily="subheading" fontWeight={400} fontSize="2xs">
+          Buy CELO
+        </Text>
+      </Stack>
+      <Stack width={"1/3"} alignItems={"center"}>
+        <Circle size="40px" bgColor={step < 2 ? "goodGrey.300" : "primary"}>
+          <Text color={step < 2 ? "goodGrey.700" : "white"}> 2 </Text>
+        </Circle>
+        <Text w="112px" color="goodGrey.700" fontFamily="subheading" fontWeight={400} fontSize="2xs">
+          We swap CELO to G$
+        </Text>
+      </Stack>
+      <Stack width={"1/3"} alignItems={"center"}>
+        <Circle size="40px" bgColor={step <= 3 ? "goodGrey.300" : "primary"}>
+          <Text color={step <= 3 ? "goodGrey.700" : "white"}> 3 </Text>
+        </Circle>
+        <Text color="goodGrey.700" fontFamily="subheading" fontWeight={400} fontSize="2xs">
+          Done
+        </Text>
+      </Stack>
+    </HStack>
+  </VStack>
+);
+
+export const Onramper = ({
+  onEvent,
+  step,
+  setStep,
+  widgetParams = { onlyCryptos: "CUSD_CELO", isAddressEditable: false },
+  targetNetwork = "CELO",
+  targetWallet
+}: {
+  onEvent?: OnramperCallback;
+  step: number;
+  setStep: (step: number) => void;
+  widgetParams?: any;
+  targetWallet?: string;
+  targetNetwork?: string;
+}) => {
+  const handleEvents = useCallback((event: WebViewMessageEvent) => {
+    console.log("onramper event from webview:", event);
+    onEvent && onEvent(event);
+  }, []);
+
+  const uri = `https://buy.onramper.com/?networkWallets=${targetNetwork}:${targetWallet}&${Object.entries(widgetParams)
+    .map(([k, v]) => `${k}=${v}`)
+    .join("&")}`;
+
+  const startStepper = useCallback(() => {
+    if (step === -1) {
+      setStep(0);
+    }
+  }, [step]);
+
+  useEffect(() => {
+    window.focus(); // first force  focus for the step animation to start properly
+  }, []);
+
+  // SO fiddle: http://jsfiddle.net/wk1yv6q3/
+  useEffect(() => {
+    const checkFocus = (e: any) => {
+      console.log("checkFocus", { e, step });
+      if (document.activeElement === document.querySelector("iframe")) {
+        startStepper();
+      } else if (step === 0) {
+        setStep(-1);
+      }
+    };
+    window.addEventListener("focus", checkFocus);
+    window.addEventListener("blur", checkFocus);
+    return () => {
+      window.removeEventListener("blur", checkFocus);
+      window.removeEventListener("focus", checkFocus);
+    };
+  }, [step]);
```

Comment:
why does it depend on step?
to me the logic doesnt look right. if the user will focus on the widget when in one of the other steps it will reset the stepper

Reply:
because it would do a reset only when step = 0 (meaning, user interacted with the frame) but then clicked away again.
It will only do a start step when step = -1

all other cases, the steps progress is not touched by this blur/focus handler

So the dep is needed else it won't update the value of step in the callback.. (right?)

Reply:
startstepper is redundant. the if hsould then be if(document... && step===-1) setStep(0)